mesonbuild/compilers/compilers.py

```diff
@@ -295,6 +296,10 @@ def get_base_compile_args(options: 'KeyedOptionDictType', compiler: 'Compiler',
                 mode=get_option_value(options, OptionKey('b_lto_mode'), 'default')))
     except (KeyError, AttributeError):
         pass
+
+    if option_enabled(compiler.base_options, options, OptionKey('b_legal_code')):
```

Comment:
@eli-schwartz This conditional is always false but I don't get why?

Reply:
You still need to add to the compiler that it suppots `b_legal_code`, say in the `GnuCompiler.__init__` method

Reply:
Although, come to think of it, GnuDCompiler inherits that, as does the out of tree GnuRustCompiler patches I have somewhere...

Reply:
Thanks, I was wondering if it was something like that but I couldn't spot it. Doh.

Reply:
So, probably have to add them to `GnuCCompiler and GnuCPPCompiler